filebeat/input/file/state.go

```diff
@@ -93,7 +93,7 @@ func (s *States) Cleanup() {
 
 	for _, state := range s.states {
 		ttl := state.TTL
-		if ttl >= 0 && currentTime.Sub(state.Timestamp) > ttl {
+		if ttl >= 0 && currentTime.Sub(state.Timestamp) >= ttl {
```

Comment:
I would be ok with `ttl==0 || currentTime.Sub(state.Timestamp) >= ttl` I had one test case where state.Timestamp == currentTimestamp ! Not sure how this was able to happen (windows ...)